lib/BloqadeGates/src/types.jl

```diff
@@ -0,0 +1,41 @@
+struct RydbergPulse{D} <: AbstractBlock{D}
+    rydberg_hamiltonian::Union{RydbergHamiltonian, RydbergHamiltonian3}
+    t_start::Float64
+    t_end::Float64
+    backend::Union{Type{KrylovEvolution}, Type{SchrodingerProblem}}
+    step::Float64
+end
+
+RydbergPulse(rh::RydbergHamiltonian3, t_start::Real, t_end::Real; backend = KrylovEvolution, step = 1e-2) = RydbergPulse{3}(rh, t_start, t_end, backend, step)
+RydbergPulse(rh::RydbergHamiltonian, t_start::Real, t_end::Real; backend = KrylovEvolution, step = 1e-2) = RydbergPulse{2}(rh, t_start, t_end, backend, step)
+RydbergPulse(rh::RydbergHamiltonian3, t::Real; backend = KrylovEvolution, step = 1e-2) = RydbergPulse{3}(rh, 0.0, t, backend, step)
+RydbergPulse(rh::RydbergHamiltonian, t::Real; backend = KrylovEvolution, step = 1e-2) = RydbergPulse{2}(rh, 0.0, t, backend, step)
+
+function YaoAPI.apply!(reg::AbstractRegister{D}, p::RydbergPulse{D}) where D
+    evo = p.backend
+    if evo === KrylovEvolution
+        @assert p.t_start <= p.t_end
+        ts = p.t_start:p.step:p.t_end
+    else
+        ts = (p.t_start, p.t_end)
+    end
+    prob = evo(reg, ts, p.rydberg_hamiltonian)
+    emulate!(prob)
+    return reg
+end
+
+Base.:(|>)(reg::AbstractRegister{D}, p::RydbergPulse{D}) where D = apply!(reg, p)
```

Comment:
This is predefined in Yao for all `AbstractBlock` already?

Reply:
You are correct. I do not remember why I defined this. I will remove it.